File: services/scraper-api/app/modules/page_metadata/manifest.py
from app.modules.base import ModuleManifest

MODULE_KEY = "page_metadata"

MANIFEST = ModuleManifest(
    key=MODULE_KEY,
    name="Page Metadata",
    description="Fetch a public URL and extract basic page metadata.",
    capabilities={
        "manual_runs": True,
        "schedules": True,
        "input_fields": [
            {
                "key": "url",
                "label": "URL",
                "type": "url",
                "required": True,
            }
        ],
    },
)

File: apps/scraper/src/modules/page-metadata/manifest.ts
import type { ScraperModule } from "@/modules/registry"

export const pageMetadataModule: ScraperModule = {
  key: "page_metadata",
  name: "Page Metadata",
  description: "Fetch public URLs and capture page metadata.",
  href: "/modules/page-metadata",
  icon: "appWindow",
}
